crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -520,10 +520,31 @@ impl<'a> BodyVisitor<'a> {
     }
 
     fn finish(self) -> BodyEntries<'a> {
+        let BodyVisitor {
+            returns,
+            yields,
+            mut raised_exceptions,
+            ..
+        } = self;
+
+        // Deduplicate exceptions collected:
+        // no need to complain twice about `raise TypeError` not being documented
+        // just because there are two separate `raise TypeError` statements in the function
+        raised_exceptions.sort_unstable_by(|left, right| {
+            left.qualified_name
+                .segments()
+                .cmp(right.qualified_name.segments())
+                .then_with(|| left.start().cmp(&right.start()))
+                .then_with(|| left.end().cmp(&right.end()))
+        });
+        raised_exceptions.dedup_by(|left, right| {
+            left.qualified_name.segments() == right.qualified_name.segments()
+        });
```

Comment:
Could we use a `BTreeSet` instead, to get this for free?

Reply:
I don't think so, because we're storing `ExceptionEntry` objects in the `Vec` rather than `QualifiedName`s. `ExceptionEntry`s have two fields — one is a `QualifiedName` and the other is a `TextRange`. We want to make sure that no two `ExceptionEntry`s in the Vec have the same `QualifiedName`, regardless of whether they have the same `TextRange` or not

Reply:
I suppose we could add this custom ordering logic to an `Ord` implementation for `ExceptionEntry` and then use a `BTreeSet` to store the `ExceptionEntry`s?

Reply:
It _is_ possible to do this by implementing `Hash` for `ExceptionEntry` manually, and omitting `TextRange`. But probably not worth it, can create confusion.